### DISCUSSION TITLE
Simplify constant definitions

### DISCUSSION THREAD

**Discussion by dozer75** [dotnet/csharplang#1113] [General]
In lots of code today, magic strings are used to in different scenarios. The `nameof` keyword greatly decreased the number of magic strings usages, simplified usage and reduced typos issues.

The `nameof` keyword also made it possible to create constant string definitions of the constant name which also is a lot of help. Constants are normally written like this:
```
public class SomeClass
{
    const string SomeConst = nameof(SomeConst);
}
```

However, I think writing constant string definitions like the above example is cumbersome, and I think that introducing a new keyword `define` constant definitions could be done in an easier format:

```
public class SomeClass
{
    define SomeConst;
}
```
Which internally should transform to a const string as defined in the previous const example.

The default access modifier for the `define` keyword should be the same as `const`.

An other option is to (as other has suggested) extending `enum`. The latter example is already covered by other issues, but the first would be nice that could be defined something like:

```
enum Consts : string
{
    SomeConst // same as const string SomeConst = nameof(SomeConst)
}
```

But I think that the latter will force developers to create unneccessary enums, especially in situations you only need a simple constant string within a class definition.

I think that this would ease a lot of const definitions for developers around the magic string issue.

**Comment by alrz** [comment]
call `ToString` on an enum would have the same effect. but it's not a constant. (https://github.com/dotnet/csharplang/issues/1028)

**Comment by dozer75** [comment]
Well, then it's just as easy creating a constant `nameof`

**Comment by MkazemAkhgary** [comment]
very nice proposal, only upvoted because of proposed enum part. (`define` keyword sounds very cheap but string enum Is very rich)

**Comment by svick** [comment]
I don't think `string` constants that contain their own name are so common to warrant their own keyword, especially not one as confusing as `define`. Consider that I would never guess that `define SomeConst;` is the same as `const string SomeConst = "SomeConst";`. Also, C# already has `#define`, which is very different.

**Comment by MkazemAkhgary** [comment]
@svick exactly, but `enum : string` looks very nice and the syntax describes what exactly it means.

**Comment by CyrusNajmabadi** [comment]
To me, this falls under the buckets of:

1. Not enough value over existing solutions.
2. Not common enough to benefit from being marginally better.
